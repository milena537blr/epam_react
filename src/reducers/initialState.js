export default {
  movies: [],
  filters: {
    text: "",
    sortBy: "rating",
    searchBy: "title"
  }
};
