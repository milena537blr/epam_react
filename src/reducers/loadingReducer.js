import * as types from "../actions/actionTypes";
import { initialState } from "./initialState";

const loadingReducer = (state = initialState.isLoading, action) => {
  switch (action.type) {
    case types.LOADING:
      return action.isLoading;
    default:
      return state;
  }
};

export { loadingReducer };
