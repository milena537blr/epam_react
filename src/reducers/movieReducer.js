import * as types from "../actions/actionTypes";
import initialState from "./initialState";

export default function movieReducer(state = initialState.movies, action) {
  switch (action.type) {
    case types.LOAD_MOVIES_SUCCESS:
      // return action.movies;
      // return Object.assign([], state, action.movies);
      console.log(Object.assign([], state, action.movies));
    case types.ADD_TODO:
      return [
        ...state,
        {
          text: action.text,
          completed: false
        }
      ];
    default:
      return state;
  }
}
