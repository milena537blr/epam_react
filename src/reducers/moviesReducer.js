import * as types from "../actions/actionTypes";
import { initialState } from "./initialState";
import { call, put, all, takeLatest } from "redux-saga/effects";
import { updateMovies, updateCurrentMovie } from "../actions/actions";

// Sagas
export function* fetchMoviesAsync() {
  const response = yield call(
    fetch,
    "http://react-cdp-api.herokuapp.com/movies"
  );
  const movies = yield response.json();

  yield put(updateMovies(movies));
}

export function* watchFetchMovies() {
  yield takeLatest(types.FETCH_MOVIES, fetchMoviesAsync);
}

export function* fetchMovieByIdAsync(action) {
  const response = yield call(
    fetch,
    `http://react-cdp-api.herokuapp.com/movies/${action.payload}`
  );
  const movie = yield response.json();

  yield put(updateCurrentMovie(movie));
}

export function* watchFetchMovieById() {
  yield takeLatest(types.FETCH_MOVIE_BY_ID, fetchMovieByIdAsync);
}

// Movies Saga
export function* moviesSaga() {
  yield all([watchFetchMovies(), watchFetchMovieById()]);
}

const moviesReducer = (state = initialState.data, action) => {
  switch (action.type) {
    case types.LOAD_MOVIES_LOADING:
      return { ...state, loading: true, error: "" };
    case types.LOAD_MOVIES_SUCCESS:
      return { ...state, loading: false, movies: action.movies };
    case types.LOAD_MOVIES_FAILURE:
      return { ...state, loading: false, error: action.error };
    default:
      return state;
  }
};

export { moviesReducer };
