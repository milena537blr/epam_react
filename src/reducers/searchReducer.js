import * as types from "../actions/actionTypes";
import initialState from "./initialState";

export default function searchReducer(state = initialState.movies, action) {
  switch (action.type) {
    case types.SET_VISIBILITY_FILTER:
      return action.filter;
    default:
      return state;
  }
}