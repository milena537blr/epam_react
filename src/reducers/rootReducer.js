import { combineReducers } from "redux";
import { movieReducer as movies } from "./movieReducer";
import { searchReducer as filters } from "./searchReducer";

const rootReducer = combineReducers({
  movies,
  filters
});

export { rootReducer };
