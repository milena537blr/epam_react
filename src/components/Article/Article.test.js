import React from "react";
import { mount } from "enzyme";
import ConnectedArticle from "./Article";
import { Provider } from "react-redux";
import { render, cleanup } from "@testing-library/react";
import { moviesReducer } from "../../reducers/moviesReducer";
import { createStore } from "redux";
import { Link, Route, Router, Switch } from "react-router-dom";
import { createMemoryHistory } from "history";

afterEach(cleanup);

function renderWithRedux(
  ui,
  { initialState, store = createStore(moviesReducer, initialState) },
  {
    route = "/film/447365",
    history = createMemoryHistory({ initialEntries: [route] })
  } = {}
) {
  return {
    ...render(
      <Provider store={store}>
        <Router history={history}>{ui}</Router>
      </Provider>
    ),
    // adding `store` to the returned utilities to allow us
    // to reference it in our tests (just try to avoid using
    // this to test implementation details).
    store,
    history
  };
}
  
const card1 = {
  id: 447365,
  release_date: "2020-05-01",
  genres: ["Action", "Adventure", "Science Fiction"],
  title: "Guardians of the Galaxy Vol. 3",
  tagline: "",
  overview: "The third film based on Marvel's Guardians of the Galaxy.",
  runtime: null
};

const card2 = {
  id: 447365,
  release_date: "2018-04-07",
  genres: ["TV Movie", "Drama"],
  title: "Paterno",
  tagline: "The greater the legend, the harder the fall.",
  overview:
    "After becoming the winningest coach in college football history, Joe Paterno is embroiled in Penn State's Jerry Sandusky sexual abuse scandal, challenging his legacy and forcing him to face questions of institutional failure regarding the victims.",
  runtime: null
};

/* test("snapshot test", () => {
  const component = mount(<Article card={card1} />);
  expect(component).toMatchSnapshot();
}); */

test("generates title, overview, date, runtime, tagline of Article", () => {
  /*   const { getByText, getByAltText, rerender } = render(
    <Article card={card1} />
  ); */

  const { getByText, getByAltText } = renderWithRedux(<ConnectedArticle />, {
    initialState: {
      data: {
        movies: [
          {
            id: 447365,
            release_date: "2020-05-01",
            genres: ["Action", "Adventure", "Science Fiction"],
            title: "Guardians of the Galaxy Vol. 3",
            tagline: "",
            overview:
              "The third film based on Marvel's Guardians of the Galaxy.",
            runtime: null
          },
          {
            id: 447365,
            release_date: "2018-04-07",
            genres: ["TV Movie", "Drama"],
            title: "Paterno",
            tagline: "The greater the legend, the harder the fall.",
            overview:
              "After becoming the winningest coach in college football history, Joe Paterno is embroiled in Penn State's Jerry Sandusky sexual abuse scandal, challenging his legacy and forcing him to face questions of institutional failure regarding the victims.",
            runtime: null
          }
        ],
        loading: false,
        error: ""
      }
    }
  },
  {
    route: '/film/447365',
  });

  expect(getByText(card1.title)).toBeInTheDocument();
  expect(getByText(card1.overview)).toBeInTheDocument();
  expect(getByText(card1.tagline)).toBeInTheDocument();
  expect(getByText(card1.release_date)).toBeInTheDocument();
  expect(getByText(card1.runtime, { exact: false })).toBeInTheDocument();
  expect(getByAltText(card1.title)).toBeInTheDocument();

  // rerender(<Article card={card2} />);

  /* expect(getByText(card2.title)).toBeInTheDocument();
  expect(getByText(card2.overview)).toBeInTheDocument();
  expect(getByText(card2.tagline)).toBeInTheDocument();
  expect(getByText(card2.release_date)).toBeInTheDocument();
  expect(getByText(card2.runtime, { exact: false })).toBeInTheDocument();
  expect(getByAltText(card2.title)).toBeInTheDocument(); */
});
