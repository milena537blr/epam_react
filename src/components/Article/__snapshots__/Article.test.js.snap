// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Article /> <Article /> snapshot test 1`] = `
<Article
  card={
    Object {
      "date": 2007,
      "description": "American drama",
      "genre": "Dramas",
      "id": 1,
      "name": "Vampire diaries",
      "text": "The Vampire Diaries is an American supernatural teen drama television series developed by Kevin Williamson and Julie Plec, based on the popular book series of the same name written by L. J. Smith.",
      "time": 145,
    }
  }
>
  <Box
    className="article"
    direction="horizontal"
    inline={false}
  >
    <div
      className="root article horizontal"
      style={
        Object {
          "height": "undefined",
          "margin": "undefined",
          "marginBottom": "undefined",
          "marginLeft": "undefined",
          "marginRight": "undefined",
          "marginTop": "undefined",
          "maxHeight": "undefined",
          "maxWidth": "undefined",
          "minHeight": "undefined",
          "minWidth": "undefined",
          "padding": "undefined",
          "paddingBottom": "undefined",
          "paddingLeft": "undefined",
          "paddingRight": "undefined",
          "paddingTop": "undefined",
          "width": "undefined",
        }
      }
    >
      <figure>
        <img
          alt="Vampire diaries"
          className="image"
          src="test-file-stub"
        />
      </figure>
      <article
        data-testid="article"
      >
        <h1
          className="name"
        >
          Vampire diaries
        </h1>
        <div
          className="description"
        >
          American drama
        </div>
        <div
          className="info"
        >
          <span
            className="date"
          >
            2007
          </span>
          <span
            className="time"
          >
            145
             min
          </span>
        </div>
        <div
          className="text"
        >
          The Vampire Diaries is an American supernatural teen drama television series developed by Kevin Williamson and Julie Plec, based on the popular book series of the same name written by L. J. Smith.
        </div>
      </article>
    </div>
  </Box>
</Article>
`;
